### ISSUE TITLE
Samba share 9.5.1 only work in eth0

### ISSUE THREAD

**Issue by Gferretta** [home-assistant/addons#2074]
Hi

With actual version do not posible to access via SAMBA over wlan0's IP

In configutarion, 
interface: ''

Also probe it:
interface: wlan0, eth0

but same case,

Any idea?

With previous version also eth0 and wlan0's IP was accesible via SAMBA  automatically

**Comment by sanderlv** [comment]
https://community.home-assistant.io/t/samba-share-9-5-1-vs-9-3-1-no-access-on-9-5-1/319515

This seems to be my exact issue also...

**Comment by sanderlv** [comment]
Do I understand correctly that since this latest update one need to "specify" the interface names" that samba needs to listrn on? If so how do I find the available interface names?

**Comment by gadget78** [comment]
its easier to just remove the `interface: ‘’` entry in configuration altogether !
if it then doesn't work, look in the log 7th line down it states what the hostname and interface its then using (in green) 
then play around specifying the interface, and check its changing etc ...

**Comment by sanderlv** [comment]
Thanks for taking the time to reply!

Ok, but 2 questions:

If I update to 9.5.1 it instantly doesn't work (while all previous versions work)
If I roll back pre-9.5.1. it all works.

If I use 9.5.1. with or without interfaces: '' is does not work.

If I want to follow your tip, I still have my question from a post above:
If so how do I find the available interface names?

Why doesn't the 9.5.1. samba addon just not keep on using "all" interface available and am I now in the situations to "search" for interfaces?!

Where is this even documented then?

**Comment by ludeeus** [comment]
`interface` was removed in 9.5.0, so on 9.5.0 and 9.5.1 this does nothing.
The add-on gets the default interface from the supervisor and binds to that.
9.5.0 had a bug where the add-on did not get any interfaces, fixing that is the only thing that changed to 9.5.1

**Comment by sanderlv** [comment]
Good to know... but now I and many others (see amount of complains in github addons section have issue that samba is not working anymore...?

**Comment by ludeeus** [comment]
It's working over the default interface

**Comment by sanderlv** [comment]
Probable that is my (and others') issue... we are not using the default interface (lan) but the wan...

**Comment by sanderlv** [comment]
And even in 9.5.1. when I configure:
interface: ztbpaltvee

I get:
[13:11:56] INFO: Using hostname=homeassistant interface=enp0s18

So it is not taking the non-default interface.

why had this to change and can it be fixed please???

**Comment by ludeeus** [comment]
> interface was removed in 9.5.0, so on 9.5.0 and 9.5.1 this does nothing.

The default interface is defined on `ha su network info`

**Comment by sanderlv** [comment]
Thanks for letting me know. But why can't it just work as before? without having to worry about default interface and if not select another (that does not work) but just be active on "all" interfaces?

I do want it to work on both LAN and WAN please...

**Comment by gadget78** [comment]
oddly that line was making my instance fail, so it must be impacting something ? .. 
of course its now utilising other methods to specify which interface to use, namely the hassio_api ?? 
when you roll back and it works, which interface is it naming then ? or does it list many ? ..

**Comment by sanderlv** [comment]
See above comment

**Comment by sanderlv** [comment]
> 
> 
> > `interface` was removed in 9.5.0, so on 9.5.0 and 9.5.1 this does nothing.
> 
> The default interface is defined on `ha su network info`

Where would I have to do this:
![image](https://user-images.githubusercontent.com/16528916/125279570-6481a080-e314-11eb-9900-bb3d4d3d3f7e.png)

**Comment by ludeeus** [comment]
you are missing a `w` in `network`

**Comment by sanderlv** [comment]
> 
> 
> you are missing a `w` in `network`

yes I saw, that was after "clear" for the screenshot, but with it also doesnt work

**Comment by ludeeus** [comment]
right... my bad, it's not a part of `su` , but it's own thing, so just `ha network info`

**Comment by sanderlv** [comment]
![image](https://user-images.githubusercontent.com/16528916/125280071-f12c5e80-e314-11eb-8a95-1535cbb3a22b.png)

**Comment by ludeeus** [comment]
where you have "primary: true" is the default interface

**Comment by sanderlv** [comment]
Ok, good to know, but the matter of the fact stays after update to 9.5.1. its not working anymore... rollback it works... I use zerotier to access it but that does not work anymore now...

**Comment by Gferretta** [comment]
Hi
My wlan0 interface is enabled, but not primary.
wlan0 Is Not more accessibled from Samba 9.5.1

And no way to activated.

**Comment by lmagyar** [comment]
Another corner case, that doesn't work with 9.5.1:
```
allow_hosts:
  - localhost
```
Then I ssh into HA, use local port forwarding on port 139 and 445 (like: L 192.168.x.x:139 localhost:139), so samba is protected with ssh.

`ha network info` says eth0 is primary, wlan0 is not connected, there are no more interfaces shown. Am I correct, that ssh is using loopback in this case?

**Comment by sanderlv** [comment]
Something is definitely broken here since 9.5.1...

**Comment by PaoloTK** [comment]
From what I understand going forward it won't be possible to ruin the Samba add-on on a non-primary interface?

Unfortunately, this is a must for my setup since some integrations break when their devices are not on the same VLAN as the primary HA interface.

**Comment by Destix** [comment]
Same problem here :(  not working on ZeroTier interface
Have to downgrade to 9.3.0
It's not easy to downgrade on HA

**Comment by sanderlv** [comment]
Can this please be kept open and resolved? It worked well until the latest update...

**Comment by Destix** [comment]
Agred. Maybe it's possible add some option to enable samba on all interfaces.

**Comment by ahmaddxb** [comment]
how do you change the primary interface

**Comment by gryzli133** [comment]
Same Problem here. Could someone write how to go to previous version? Thanks in advance!

**Comment by lmagyar** [comment]
Noting changed, Samba is still unusable from other than default interface (like wlan0, loopback, etc.)

**Comment by skotekar** [comment]
Not sure if this is going to help, but after I removed the interface option from config, the shares started working for me.

Another weird thing for me was I could access the share using the IP or using \\homeassistant.local but it did does not work when using \\homeassistant

**Comment by nimamahmoudi** [comment]
I am having the same issue and removing the interface from the config did not help. Any suggested fixes yet?

**Comment by skotekar** [comment]
If you are trying to access the share from windows, it might be that an old/incorrect password is saved in windows credential cache. Make sure it is the right one (see the screenshot below). Or you can try deleting any saved windows credentials for homeassistant and then try accessing the share again.

![image](https://user-images.githubusercontent.com/628453/149847042-269aad32-aeb9-4f48-b123-228459ab5427.png)

**Comment by nimamahmoudi** [comment]
Thanks @skotekar, but it is not windows, I check the TCP port 445 using netcat (`nc SERVER_IP 445 -vvv`) and it worked perfectly fine on one interface and failed with connection error on the other interface. I also tried with windows and I got the same results, works on one interface, but fails on the other. It is also shown on the add-on logs (only lists one interface).

**Comment by lmagyar** [comment]
Half year passed and nothing changed.

I've forked the add-on and restored the `interface` option with a twist:
- it is renamed to `interfaces` (note the plural), a list
- it is backward compatible with 9.5.1, ie. it's optional, if it is not configured, behaves exactly the same as 9.5.1
- if you specify `interfaces`, it works as expected, runs on the specified interface, or interfaces
  - yes, it was possible to specify multiple interfaces in 9.4.0, as a space separated string, but specifying multiple values are more elegant with a list

Sample config, can be added to the original configuration:
```yaml
interfaces:
  - wlan0
```

How to install:
- Disable the current Samba add-on
  - `Stop` it
  - `Start on boot`: disable
  - `Watchdog`: disable
  - `Auto update`: disable
- Install forked version
  - Navigate in your Home Assistant frontend to **Supervisor** -> **Add-on Store**.
  - Click **Repositories** in the **...** menu at the top right corner, add `https://github.com/lmagyar/homeassistant-addon-samba-interface` as repository.
  - Find the "Samba share - with interfaces configuration" add-on and click it. If it doesn't show up, wait until HA refreshes the information about the add-on, or click **Reload** in the **...** menu at the top right corner.
  - Click on the "INSTALL" button.
- Copy the configuration from the original add-on
- Add the necessary `interfaces` values
- `Start`

If it solves the problems, I will create a PR to the official add-on, but it's better to know in advance, that it works on other machines than mine.

**Comment by Destix** [comment]
default interface name to zerotier is zt7nnnuehy
```
interfaces:
  - wlan0
  - eth0
  - zt7nnnuehy
```

**Comment by lmagyar** [comment]
@Destix Could you please try the new 9.6.0.1 version of this forked add-on?
- it contains the suggested changes from the core team on my PR
- it runs on all supported interfaces
- it does not require `interfaces:` config, it ignores it, you can delete it from the config
- **the big question: will it run smoothly on the Tailscale interface?**

**Comment by frenck** [comment]
> the big question: will it run smoothly on the Tailscale interface?

It won't. As that is not managed by the Supervisor.

**Comment by lmagyar** [comment]
Hmmm, how can we handle this situation?

It seems to be a valid case to access an off the grid system through a mobile stick, in that case you need a vpn to get through the mobile operators cg-nat. vpn/tailscale + ssh port forward??? Note: I'm not familiar with the supervisor's internals.

**Comment by frenck** [comment]
> Hmmm, how can we handle this situation?

We can't as it isn't controlled by Home Assistant. If one is in need of more complex setups like this; we have other installation methods available that do offer the level of customization for those cases.

**Comment by Destix** [comment]
@lmagyar 

New version does not work.

<details>
  <summary>Addon log:</summary>
```
interpret_interface: Can't find address for eth0
added interface wlan0 ip=2001:470:6328:0:8a:c1f1:73fd:60b3 bcast= netmask=ffff:ffff:ffff:ffff::
added interface wlan0 ip=192.168.3.20 bcast=192.168.3.255 netmask=255.255.255.0
making subnet name:192.168.3.20 Broadcast address:192.168.3.255 Subnet mask:255.255.255.0
create_subnets: ignoring non IPv4 interface.
making subnet name:UNICAST_SUBNET Broadcast address:0.0.0.0 Subnet mask:0.0.0.0
making subnet name:REMOTE_BROADCAST_SUBNET Broadcast address:0.0.0.0 Subnet mask:0.0.0.0
daemon_ready: daemon 'nmbd' finished starting up and ready to serve connections
interpret_interface: Can't find address for eth0
added interface wlan0 ip=2001:470:6328:0:8a:c1f1:73fd:60b3 bcast= netmask=ffff:ffff:ffff:ffff::
added interface wlan0 ip=192.168.3.20 bcast=192.168.3.255 netmask=255.255.255.0
INFO: Profiling support unavailable in this build.
```
</details>

<details>
  <summary>Addon docker:</summary>
```
docker exec -it 92d8ccbf57be /bin/bash

bash-5.1# ifconfig

docker0   Link encap:Ethernet  HWaddr 02:42:BC:09:B4:DB
          inet addr:172.17.0.1  Bcast:172.17.255.255  Mask:255.255.0.0
          inet6 addr: fe80::42:bcff:fe09:b4db/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:56187 errors:0 dropped:0 overruns:0 frame:0
          TX packets:2593593 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:6063998 (5.7 MiB)  TX bytes:125073939 (119.2 MiB)

eth0      Link encap:Ethernet  HWaddr DC:A6:32:0F:33:62
          UP BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)

hassio    Link encap:Ethernet  HWaddr 02:42:5C:23:AB:B1
          inet addr:172.30.32.1  Bcast:172.30.33.255  Mask:255.255.254.0
          inet6 addr: fe80::42:5cff:fe23:abb1/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:7096278 errors:0 dropped:0 overruns:0 frame:0
          TX packets:10720396 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:1551846760 (1.4 GiB)  TX bytes:3518155388 (3.2 GiB)

lo        Link encap:Local Loopback
          inet addr:127.0.0.1  Mask:255.0.0.0
          inet6 addr: ::1/128 Scope:Host
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:7115239 errors:0 dropped:0 overruns:0 frame:0
          TX packets:7115239 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:1077301144 (1.0 GiB)  TX bytes:1077301144 (1.0 GiB)

veth14a0764 Link encap:Ethernet  HWaddr 7E:B7:01:B6:20:55
          inet6 addr: fe80::7cb7:1ff:feb6:2055/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:58279 errors:0 dropped:0 overruns:0 frame:0
          TX packets:3511172 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:4122117 (3.9 MiB)  TX bytes:270328309 (257.8 MiB)

veth3485367 Link encap:Ethernet  HWaddr 52:AE:2E:A6:11:97
          inet6 addr: fe80::50ae:2eff:fea6:1197/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:666588 errors:0 dropped:0 overruns:0 frame:0
          TX packets:3986798 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:202642440 (193.2 MiB)  TX bytes:473475579 (451.5 MiB)

veth58686c3 Link encap:Ethernet  HWaddr 72:DE:E6:28:98:EB
          inet6 addr: fe80::70de:e6ff:fe28:98eb/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:3436474 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:0 (0.0 B)  TX bytes:264441227 (252.1 MiB)

veth5be2f1d Link encap:Ethernet  HWaddr 36:3E:94:51:78:2E
          inet6 addr: fe80::343e:94ff:fe51:782e/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:665 errors:0 dropped:0 overruns:0 frame:0
          TX packets:1476593 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:304735 (297.5 KiB)  TX bytes:108488796 (103.4 MiB)

veth5f7829c Link encap:Ethernet  HWaddr E6:E3:B7:D5:53:E4
          inet6 addr: fe80::e4e3:b7ff:fed5:53e4/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:2119240 errors:0 dropped:0 overruns:0 frame:0
          TX packets:6704431 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:714270344 (681.1 MiB)  TX bytes:1156876684 (1.0 GiB)

veth71549c1 Link encap:Ethernet  HWaddr EE:83:EE:E2:9F:71
          inet6 addr: fe80::ec83:eeff:fee2:9f71/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:447749 errors:0 dropped:0 overruns:0 frame:0
          TX packets:3935534 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:31232525 (29.7 MiB)  TX bytes:301215417 (287.2 MiB)

veth732f0d6 Link encap:Ethernet  HWaddr 1A:B5:94:80:8E:72
          inet6 addr: fe80::18b5:94ff:fe80:8e72/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:4667307 errors:0 dropped:0 overruns:0 frame:0
          TX packets:8049403 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:990295835 (944.4 MiB)  TX bytes:2718854029 (2.5 GiB)

veth770d65a Link encap:Ethernet  HWaddr 52:03:E6:DA:EB:79
          inet6 addr: fe80::5003:e6ff:feda:eb79/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:56187 errors:0 dropped:0 overruns:0 frame:0
          TX packets:2594086 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:6850616 (6.5 MiB)  TX bytes:125110311 (119.3 MiB)

veth8f0abe9 Link encap:Ethernet  HWaddr 22:EC:27:F0:F8:A6
          inet6 addr: fe80::20ec:27ff:fef0:f8a6/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:526138 errors:0 dropped:0 overruns:0 frame:0
          TX packets:3128796 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:38610383 (36.8 MiB)  TX bytes:251303744 (239.6 MiB)

vethcb55dfe Link encap:Ethernet  HWaddr 66:8B:3E:D1:C0:B0
          inet6 addr: fe80::648b:3eff:fed1:c0b0/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:39 errors:0 dropped:0 overruns:0 frame:0
          TX packets:3436462 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:3857 (3.7 KiB)  TX bytes:264441523 (252.1 MiB)

vethd93985b Link encap:Ethernet  HWaddr 2A:62:96:F1:FC:F4
          inet6 addr: fe80::2862:96ff:fef1:fcf4/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:438 errors:0 dropped:0 overruns:0 frame:0
          TX packets:3436846 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:51415 (50.2 KiB)  TX bytes:264565217 (252.3 MiB)

vethdb2e696 Link encap:Ethernet  HWaddr C6:54:E9:E4:B0:B3
          inet6 addr: fe80::c454:e9ff:fee4:b0b3/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:67 errors:0 dropped:0 overruns:0 frame:0
          TX packets:3436530 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:8098 (7.9 KiB)  TX bytes:264453534 (252.2 MiB)

vethdd5b0cf Link encap:Ethernet  HWaddr 0E:32:C0:F0:87:3F
          inet6 addr: fe80::c32:c0ff:fef0:873f/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:3436227 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:0 (0.0 B)  TX bytes:264412085 (252.1 MiB)

vethee81539 Link encap:Ethernet  HWaddr 32:C8:43:D5:40:3A
          inet6 addr: fe80::30c8:43ff:fed5:403a/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:105236 errors:0 dropped:0 overruns:0 frame:0
          TX packets:3517424 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:6863223 (6.5 MiB)  TX bytes:269729659 (257.2 MiB)

wlan0     Link encap:Ethernet  HWaddr DC:A6:32:0F:33:63
          inet addr:192.168.3.20  Bcast:192.168.3.255  Mask:255.255.255.0
          inet6 addr: 2001:470:6328:0:8cca:c1f1:73fd:60b3/64 Scope:Global
          inet6 addr: fe80::f9f7:3dea:53b0:4a65/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:33887356 errors:0 dropped:4 overruns:0 frame:0
          TX packets:25152487 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:10592794214 (9.8 GiB)  TX bytes:3256320079 (3.0 GiB)

zt7nnnuehy Link encap:Ethernet  HWaddr 62:B1:4B:37:34:61
          inet addr:10.10.3.20  Bcast:10.10.255.255  Mask:255.255.0.0
          inet6 addr: fe80::60b1:4bff:fe37:3461/64 Scope:Link
          inet6 addr: fcae:894c:ad6f:3ac:faf3::1/40 Scope:Global
          UP BROADCAST RUNNING MULTICAST  MTU:2800  Metric:1
          RX packets:8392 errors:0 dropped:0 overruns:0 frame:0
          TX packets:3731 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:1076255 (1.0 MiB)  TX bytes:457997 (447.2 KiB)
```
</details>

**Comment by lmagyar** [comment]
@Destix I've merged the changes from the PR on official samba and **restored** the interfaces configuration

I leave my fork available, and will merge changes from the official version, because I also plan to use Tailscale soon on a remote installation and I don't know how the official Samba version will work with a Tailscale + SSH config. Those who don't tunnel Samba through SSH, as I understand, will need this interfaces configuration possibility. 

Maybe others (like #1716) can play with Samba config values to solve other issues.

**Comment by Gferretta** [comment]
> Half year passed and nothing changed.
> 
> I've forked the add-on and restored the `interface` option with a twist:
> - it is renamed to `interfaces` (note the plural), a list
> - it is backward compatible with 9.5.1, ie. it's optional, if it is not configured, behaves exactly the same as 9.5.1
> - if you specify `interfaces`, it works as expected, runs on the specified interface, or interfaces
>   - yes, it was possible to specify multiple interfaces in 9.4.0, as a space separated string, but specifying multiple values are more elegant with a list
> 
> Sample config, can be added to the original configuration:
> ```yaml
> interfaces:
>   - wlan0
> ```
> 
> How to install:
> - Disable the current Samba add-on
>   - `Stop` it
>   - `Start on boot`: disable
>   - `Watchdog`: disable
>   - `Auto update`: disable
> - Install forked version
>   - Navigate in your Home Assistant frontend to **Supervisor** -> **Add-on Store**.
>   - Click **Repositories** in the **...** menu at the top right corner, add `https://github.com/lmagyar/homeassistant-addon-samba-interface` as repository.
>   - Find the "Samba share - with interfaces configuration" add-on and click it. If it doesn't show up, wait until HA refreshes the information about the add-on, or click **Reload** in the **...** menu at the top right corner.
>   - Click on the "INSTALL" button.
> - Copy the configuration from the original add-on
> - Add the necessary `interfaces` values
> - `Start`
> 
> ---
> 
> **If it solves the problems, I will create a PR to the official add-on, but it's better to know in advance, that it works on other machines than mine!**

Hi
Any News if your version is working with many interfaces?
Thanks

**Comment by lmagyar** [comment]
TLDR: It works. :+1:

Official add-on v9.6.0 will run on **all supported** interfaces, not only on the default interface. But **all supported** is a shorter list than **all** interfaces.

My fork adds the optional possibility to define an `interfaces` list to run samba on even more interfaces. If the list is not specified, it defaults to the official version, if it is specified, runs on the specified list.

**Comment by Gferretta** [comment]
> TLDR: It works. :+1:
> 
> Official add-on v9.6.0 will run on **all supported** interfaces, not only on the default interface. But **all supported** is a shorter list than **all** interfaces.
> 
> My fork adds the optional possibility to define an `interfaces` list to run samba on even more interfaces. If the list is not specified, it defaults to the official version, if it is specified, runs on the specified list.

Hi
I'have yet 9.5.1 in an HA 2022.02.03,  sup 2022.01.1 and OS 7.3
🤷‍♂️

Any idea when will release update to 9.6.0?

Thanks

**Comment by Destix** [comment]
use this solution --> https://github.com/home-assistant/addons/issues/2074#issuecomment-1015824912 <-- it works

**Comment by kds69** [comment]
`use this solution --> https://github.com/home-assistant/addons/issues/2074#issuecomment-1015824912 <-- it works`
unfortunately it doesn't fix the issue for me
samba can't find interface(s)

**Comment by lmagyar** [comment]
@kds69 

> unfortunately it doesn't fix the issue for me samba can't find interface(s)

Could you please copy/send the samba add-on log and config? What is the installation type? The add-on simply copies the interface values to the samba config file: this copy can be wrong (though I've tested a lot of corner cases), or the network config for the samba container has some issues.

**Comment by Gferretta** [comment]
> Half year passed and nothing changed.
> 
> I've forked the add-on and restored the `interface` option with a twist:
> 
> * it is renamed to `interfaces` (note the plural), a list
> * UPDATED: it is backward compatible with ~v9.5.1~ v9.6.0, ie. it's optional, if it is not configured, behaves exactly the same as ~v9.5.1~ v9.6.0 (v9.6.0 runs on **all supported** interfaces, but **all supported** is a shorter list than **all** interfaces; for reference v9.5.0 (and lower) runs on **all** interfaces, v9.5.1 runs only on the **default** interface)
> * if you specify `interfaces`, it works as expected, runs on the specified interface, or interfaces
>   
>   * yes, it was possible to specify multiple interfaces in 9.4.0, as a space separated string, but specifying multiple values are more elegant with a list
> 
> Sample config, can be added to the original configuration:
> 
> ```yaml
> interfaces:
>   - wlan0
> ```
> 
> How to install:
> 
> * Disable the current Samba add-on
>   
>   * `Stop` it
>   * `Start on boot`: disable
>   * `Watchdog`: disable
>   * `Auto update`: disable
> * Install forked version
>   
>   * Navigate in your Home Assistant frontend to **Supervisor** -> **Add-on Store**.
>   * Click **Repositories** in the **...** menu at the top right corner, add `https://github.com/lmagyar/homeassistant-addon-samba-interface` as repository.
>   * Find the "Samba share - with interfaces configuration" add-on and click it. If it doesn't show up, wait until HA refreshes the information about the add-on, or click **Reload** in the **...** menu at the top right corner.
>   * Click on the "INSTALL" button.
> * Copy the configuration from the original add-on
> * Add the necessary `interfaces` values
> * `Start`
> 
> **If it solves the problems, I will create a PR to the official add-on, but it's better to know in advance, that it works on other machines than mine!**

Works like a charm!
All interfaces ON and accessible.

Thanks

PS: My official SAMBA addon still on version 9.5.1.... any idea?

**Comment by lmagyar** [comment]
Official release: PR's CI is green now. I made some shell script mess previously, that delayed the PR's merge with 1-2 weeks, that's my fault, sorry. :(

**Comment by Destix** [comment]
> `use this solution --> https://github.com/home-assistant/addons/issues/2074#issuecomment-1015824912 <-- it works` unfortunately it doesn't fix the issue for me samba can't find interface(s)

1. Log into samba docker. (something like `docker exec -it 92d8ccbf57be /bin/bash` )
2. List all interfaces. `ifconfig`
3. Add needed interfaces to add-on config

**Comment by kds69** [comment]
@[lmagyar](https://github.com/lmagyar)

> @kds69
> 
> > unfortunately it doesn't fix the issue for me samba can't find interface(s)
> 
> Could you please copy/send the samba add-on log and config? What is the installation type? The add-on simply copies the interface values to the samba config file: this copy can be wrong (though I've tested a lot of corner cases), or the network config for the samba container has some issues.

...giving it a second try, it ran through interfaces as expected, failed on eth0 and succeeded on wlan0.
so it works !!
(not sure what I have changed in the meantime ?! but it seems to work flawlessly. I have stroke out my initial comment!)

In parallel, I am exploring other solutions as well since I realize through your solution that my HA installation, which is following the recommendation to have an ipstatic for ethernet,  is indeed forcing to same ipstatic and address for the wifi interface.
I am now trying to move wlan0 to another ipstatic and see if it helps with 9.5.1, as I prefer to have all full ethernet including samba.

**Comment by lmagyar** [comment]
@kds69 Note: 9.5.1 will run only on 1 interface, the "default" (usually eth0 or wlan0, depending on your configuration).

**Comment by kds69** [comment]
> @kds69 Note: 9.5.1 will run only on 1 interface, the "default" (usually eth0 or wlan0, depending on your configuration).

Sure but so far 9.5.1 was failing to find eth0. 
A way to fix it is your PR which is anyway a bullet-proof approach to my view

Another is to fix the interfaces overlap in Hassio (according to my limited tests), achieve thanks to below insights on how to manipulate hassio interfaces:
[quote="Guide: Connecting Pi with Home Assistant OS to wifi (or other networking changes), post:1, topic:98768"]
Easiest method for Wifi (or setting Static IP) on HassOS If you are having problems creating a working network keyfile or importing the keyfile, this alternate method might work for you. It could also be that you have some other problem so this is not a “cure-all” for your wifi/network issues. NOTE: This method assumes you already have a working ethernet connection. Also, wifi is much more unreliable than ethernet and not recommended at all, but if you have no other choice… The short version:
[/quote]

**Comment by lmagyar** [comment]
Not stale: there is already a PR waiting to be merged that solves this issue.

**Comment by lmagyar** [comment]
The PR is still waiting to be merged.

**Comment by outofsight** [comment]
I have problems using SMB over zerotier. This forked addon can solve?

**Comment by lmagyar** [comment]
Maybe. Try the forked version with adding this to the config:
```
interfaces: []
```
This will enable Samba on all interfaces, then you can filter down to the necessary minimum interface list.

**Comment by outofsight** [comment]
> Maybe. Try the forked version with adding this to the config:
> 
> ```
> interfaces: []
> ```
> 
> This will enable Samba on all interfaces, then you can filter down to the necessary minimum interface list.

I can confirm that with this forked addon I can access SMB shares over ZeroTier. Thanks

Any reason the main addon block this kind of service? There is some concern I should know before enabling this?

**Comment by aihngel** [comment]
> > Maybe. Try the forked version with adding this to the config:
> > ```
> > interfaces: []
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > This will enable Samba on all interfaces, then you can filter down to the necessary minimum interface list.
> 
> I can confirm that with this forked addon I can access SMB shares over ZeroTier. Thanks
> 
> Any reason the main addon block this kind of service? There is some concern I should know before enabling this?

Unfortunately this is the same kind of behavior that is driving people away from HA.  I've had to go back to ubuntu+HA due to hassOS removing the ability to use the docker command thereby blocking deconz otau file from being moved into a container.  I too have this same issue with Samba.  The fact that they wont let it autodiscover or be configurable anymore is completely pointless.

**Comment by SamJongenelen** [comment]
FWIW: I've been using this community addon with more success: 
https://github.com/dianlight/hassio-addons/tree/master/sambanas
